openmls/src/group/core_group/proposals.rs

```diff
@@ -167,6 +167,38 @@ impl QueuedProposal {
     }
 }
 
+/// Helper struct to collect proposals such that they are unique and can be read
+/// out in the order in that they were added.
+struct OrderedProposalRefs {
```

Comment:
Maybe we could use `indexmap::IndexSet` for this? Might help reducing allocations - We're very probably already using `indexmap` somewhere deep in our dependencies so we'd get this for free